infrastructure/modules/observability/variables.tf

```diff
@@ -24,13 +24,13 @@ variable "prefix" {
 
 variable "azurerm_resource_group_obs_name" {
   type        = string
-  default     = ""
+  default     = null
   description = "Optional explicit name of the observability resource group"
 }
 
 variable "log_analytics_workspace_name" {
   type        = string
-  default     = ""
+  default     = null
   description = "Name for the Log Analytics workspace."
```

Comment:
Is this description misleading now? I think it would be more correct to say that you specify this if you want to BYO resource. The script does not create the resource if this is set.